packages/shared-ux/chrome/navigation/src/navigation.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiAccordion, EuiLink, EuiSpacer, EuiText } from '@elastic/eui';
+import React from 'react';
+import { ILocatorDefinition, NavigationProps, NavItem } from '../types';
+import { useServices } from './services';
+
+export const Navigation = (props: NavigationProps) => {
+  const { id, items: navItems, highlightedItem, initiallyOpenSections } = props;
+
+  const services = useServices();
+
+  const { getLocator } = services;
+
+  const handleClick = (locator: ILocatorDefinition | undefined) => () => {
+    if (locator) {
+      const locatorInstance = getLocator(locator.id); // FIXME: use local cache rather than lookup each time
```

Comment:
I don't think you need to cache this, as lookup from cache should not be faster than lookup from registry.

```suggestion
      const locatorInstance = getLocator(locator.id);
```